kaitaistruct.py

```diff
@@ -335,30 +336,35 @@ def bytes_terminate(data, term, include_term):
 
     @staticmethod
     def process_xor_one(data, key):
+        if key == 0:
+            return data
+
         if PY2:
             return bytes(bytearray(v ^ key for v in bytearray(data)))
         else:
             return bytes(v ^ key for v in data)
 
     @staticmethod
     def process_xor_many(data, key):
+        if key == b'\x00' * len(key):
```

Comment:
Overhead on every call.

Reply:
If the key is few bytes and data is millions of bytes, then this check has negligible cost. However... you are right. It might be better to use something like `not any(key)` which would not read entire key but only up to first non-zero byte.

Reply:
You are also right. So maybe check 2 sizes and take decision based on them?